tests/common/plugins/ptfadapter/ptfadapter.py

```diff
@@ -104,16 +106,18 @@ def _init_ptf_dataplane(self, ptf_ip, ptf_nn_port, device_num, ptf_port_set, ptf
         for id, ifname in ptf.config['port_map'].items():
             device_id, port_id = id
             ptf.dataplane_instance.port_add(ifname, device_id, port_id)
-
+        self.connected = True
```

Comment:
The above loop is adding multiple ports, will there be a case where adding ports failed in the middle? In this case, we might still want to delete whatever added ports?

Reply:
It's a good catch. I read though the ptf code, and confirmed that no exception will be thrown during ```port_add```.  So the ```for``` loop won't break until loop end. In other words, the added ports will always be cleared.